src/scheduling/appointments/edit/EditAppointment.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory, useParams } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { Spinner, Button } from '@hospitalrun/components'
+import { isBefore } from 'date-fns'
+
+import AppointmentDetailForm from '../AppointmentDetailForm'
+import useTitle from '../../../page-header/useTitle'
+import Appointment from '../../../model/Appointment'
+import { updateAppointment, fetchAppointment } from '../appointment-slice'
+import { RootState } from '../../../store'
+
+const EditAppointment = () => {
+  const { t } = useTranslation()
+  useTitle(t('scheduling.appointments.editAppointment'))
+  const history = useHistory()
+  const dispatch = useDispatch()
+
+  const [appointment, setAppointment] = useState({} as Appointment)
+  const [errorMessage, setErrorMessage] = useState('')
+  const { appointment: reduxAppointment, patient, isLoading } = useSelector(
+    (state: RootState) => state.appointment,
+  )
+
+  useEffect(() => {
+    setAppointment(reduxAppointment)
+  }, [reduxAppointment])
+
+  const { id } = useParams()
+  useEffect(() => {
+    if (id) {
+      dispatch(fetchAppointment(id))
+    }
+  }, [id, dispatch])
+
+  const onCancel = () => {
+    history.push(`/appointments/${appointment.id}`)
+  }
+
+  const onSave = () => {
+    let newErrorMessage = ''
+    if (isBefore(new Date(appointment.endDateTime), new Date(appointment.startDateTime))) {
+      newErrorMessage += ` ${t('scheduling.appointment.errors.startDateMustBeBeforeEndDate')}`
+    }
+
+    if (newErrorMessage) {
+      setErrorMessage(newErrorMessage.trim())
+      return
+    }
```

Comment:
This code is duplicated with the New Appointment save logic. 

It seems like we should be able to use the same logic in both places.

Reply:
Yes, but I'm not sure where it should go. It's the same for New Patient / Edit Patient.

Reply:
I'm not sure either, right now. Let's think on this and we can figure out the right abstraction later.

Reply:
Once we have two complete modules with new/view/edit/delete, maybe we should make a point to look over the code in general with an eye for code that's duplicated as I'm sure there are a few more instances. Just now I can think of the Save/Cancel buttons, Appointment and Patient objects used in tests, getFriendlyId, form field change handlers...

Reply:
I agree! We're getting close to that point, if not at that point when this PR gets merged.